src/operations/execute_operation.ts

```diff
@@ -40,24 +40,32 @@ export function executeOperation<
   T extends OperationBase<TOptions, TResult>,
   TOptions = OptionsFromOperation<T>,
   TResult = ResultTypeFromOperation<T>
->(topology: Topology, operation: T): Promise<TResult>;
+>(topology: Topology | undefined, operation: T): Promise<TResult>;
 export function executeOperation<
   T extends OperationBase<TOptions, TResult>,
   TOptions = OptionsFromOperation<T>,
   TResult = ResultTypeFromOperation<T>
->(topology: Topology, operation: T, callback: Callback<TResult>): void;
+>(topology: Topology | undefined, operation: T, callback: Callback<TResult>): void;
 export function executeOperation<
   T extends OperationBase<TOptions, TResult>,
   TOptions = OptionsFromOperation<T>,
   TResult = ResultTypeFromOperation<T>
->(topology: Topology, operation: T, callback?: Callback<TResult>): Promise<TResult> | void;
+>(
+  topology: Topology | undefined,
+  operation: T,
+  callback?: Callback<TResult>
+): Promise<TResult> | void;
 export function executeOperation<
   T extends OperationBase<TOptions, TResult>,
   TOptions = OptionsFromOperation<T>,
   TResult = ResultTypeFromOperation<T>
->(topology: Topology, operation: T, callback?: Callback<TResult>): Promise<TResult> | void {
+>(
+  topology: Topology | undefined,
+  operation: T,
+  callback?: Callback<TResult>
+): Promise<TResult> | void {
   if (topology == null) {
-    throw new TypeError('This method requires a valid topology instance');
+    throw new MongoClientClosedError();
   }
```

Comment:
It may be a point of discussion whether this undefined topology check should happen here or where `executeOperation` is actually called. This way significantly reduces the number of checks required for this change. One downside is that we can't specifically identify the method that caused the `MongoClientClosedError` in the error message.

Another question is whether we want to be throwing the error in this way or using `maybePromise` to use the callback.